mypy/meet.py

```diff
@@ -21,6 +21,19 @@
 
 def meet_types(s: Type, t: Type) -> ProperType:
     """Return the greatest lower bound of two types."""
+    if (isinstance(s, TypeAliasType) and isinstance(t, TypeAliasType) and
+            s.is_recursive and t.is_recursive):
+        # This case can trigger an infinite recursion, general support for this will be
```

Comment:
Is it worth breaking this logic out into a function? (Not sure.) Similar question join I guess